charts/trino/templates/tests/test-jmx.yaml

```diff
@@ -40,28 +62,38 @@ metadata:
     {{- include "trino.labels" . | nindent 4 }}
     app.kubernetes.io/component: test
     test: jmx
+  annotations:
+    "helm.sh/hook": test
+    "helm.sh/hook-delete-policy": hook-succeeded
 data:
   test.py: |
     from urllib.request import urlopen
     from urllib.error import URLError, HTTPError
     import json
+    import logging
+    import sys
+    import time
 
-    url = "http://prometheus-operator-kube-p-prometheus:9090/api/v1/targets?scrapePool=serviceMonitor/{{ .Release.Namespace }}/{{ include "trino.fullname" . }}/0&state=active"
+    logger = logging.getLogger(__name__)
+    target_service = sys.argv[1]
+    url = f"http://prometheus-operator-kube-p-prometheus:9090/api/v1/targets?scrapePool=serviceMonitor/{{ .Release.Namespace }}/{target_service}/0&state=active"
     while True:
       try:
         with urlopen(url) as response:
           data = json.load(response)
       except (URLError, HTTPError) as e:
-        print("Error fetching targets, Prometheus service might not be ready: ", e)
-        continue
+          logger.warning("Error fetching targets, Prometheus service might not be ready: ", e)
+          time.sleep(10)  # Retry after 10 seconds
```

Comment:
How about only 2 seconds? 10 seems like a lot. I haven't added this initially since I expected Prometheus to get ready fairly quickly, but I agree it's spamming logs, especially if something goes wrong.

Reply:
2 seconds seams reasonable :)